src/api.js

```diff
@@ -0,0 +1,797 @@
+
+**Warning**: A statement belonging to a database that has been closed cannot
+be used anymore.
+
+Databases **must** be closed, when you're finished with them, or the
+memory consumption will grow forever
+ */
+Database.prototype["close"] = function() {
+    var _, func, ref, ref1, stmt;
+    ref = this.statements;
+    for (_ in ref) {
+        stmt = ref[_];
+        stmt["free"]();
+    }
+    ref1 = this.functions;
+    for (_ in ref1) {
+        func = ref1[_];
+        removeFunction(func);
+    }
+    this.functions = {};
+    this.handleError(sqlite3_close_v2(this.db));
+    FS.unlink("/" + this.filename);
+    return this.db = null;
+};
+
+/* Analyze a result code, return null if no error occured, and throw
+an error with a descriptive message otherwise
+@nodoc
+ */
+Database.prototype.handleError = function(returnCode) {
+    var errmsg;
+    if (returnCode === SQLITE_OK) {
+        return null;
+    } else {
+        errmsg = sqlite3_errmsg(this.db);
+        throw new Error(errmsg);
+    }
+};
+
+/* Returns the number of rows modified, inserted or deleted by the
+most recently completed INSERT, UPDATE or DELETE statement on the
+database Executing any other type of SQL statement does not modify
+the value returned by this function.
+
+@return [Number] the number of rows modified
+ */
+Database.prototype["getRowsModified"] = function() {
+    return sqlite3_changes(this.db);
+};
+
+/* Register a custom function with SQLite
+@example Register a simple function
+    db.create_function("addOne", function(x) {return x+1;})
+    db.exec("SELECT addOne(1)") // = 2
+
+@param name [String] the name of the function as referenced in SQL statements.
+@param func [Function] the actual function to be executed.
+ */
+Database.prototype["create_function"] = function(name, func) {
+    var func_ptr, wrapped_func;
+    wrapped_func = function(cx, argc, argv) {
+        var arg, args, blobptr, data_func, error, i, k, ref, result, value_ptr, value_type;
+        args = [];
+        for (i = k = 0, ref = argc; 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
+            value_ptr = getValue(argv + (4 * i), "i32");
+            value_type = sqlite3_value_type(value_ptr);
+            data_func = (function() {
+                switch (false) {
+                case value_type !== 1:
+                    return sqlite3_value_double;
+                case value_type !== 2:
+                    return sqlite3_value_double;
+                case value_type !== 3:
+                    return sqlite3_value_text;
+                case value_type !== 4:
+                    return function(ptr) {
+                        var blob_arg, blob_ptr, j, l, ref1, size;
+                        size = sqlite3_value_bytes(ptr);
+                        blob_ptr = sqlite3_value_blob(ptr);
+                        blob_arg = new Uint8Array(size);
+                        for (j = l = 0, ref1 = size; 0 <= ref1 ? l < ref1 : l > ref1; j = 0 <= ref1 ? ++l : --l) {
```

Comment:
we can iterate in increasing order

Reply:
implemented suggestion in commit c10fb519998c58b2272c88f12bb1b57d3241f190